src/bin/pgcopydb/pgcmd.c

```
@@ -897,6 +903,13 @@ pg_restore_db(PostgresPaths *pgPaths,
 	args[argsIndex++] = "--dbname";
 	args[argsIndex++] = (char *) connStrings->safeTargetPGURI.pguri;
 
+	const char *sectionOption = postgresDumpSectionToString(options.section);
+	if (sectionOption != NULL)
+	{
+		args[argsIndex++] = "--section";
+		args[argsIndex++] = (char *) sectionOption;
+	}
```

That needs an else branch that just returns false, right?

---

That sounds like a silent error to me. We only return NULL when we detected a bug of some sorts... (calling the function with an unexpected value). I do not think it makes sense to act as if we didn't notice the problem.

---

what should be the behavior then here when the value is NULL?

---

Error message and failure of the whole process. NULL here means we know we have a bug, we don't know the section that was asked to be taken care of.

---

I have introduced another enum to represent the restore options.